datafusion-cli/src/catalog.rs

```diff
@@ -166,3 +194,50 @@ impl SchemaProvider for DynamicFileSchemaProvider {
         self.inner.table_exist(name)
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use datafusion::prelude::SessionContext;
+
+    #[tokio::test]
+    async fn query_http_location_test() -> Result<()> {
```

Comment:
please add tests with malformed, empty url

Reply:
Do you have any suggestions? As far as I can tell, until I actually run the query it does not check the validity of the url, all it cares that it starts with `http:` or `https:`, and if it doesn't start with any of those then the test will not be touching the code I wrote. I could actually run the query but I would need to point it to a parquet file somewhere on the web which would no longer be a unit test but more an integration test. More importantly, I would need to find a parquet file which is always available which I don't have at the moment

Reply:
I agree that this is sufficient coverage --i don't think it would be good to require internet access to run the tests successfully either.